pkg/build/builder/cmd/builder.go

```diff
@@ -116,7 +116,7 @@ func (c *builderConfig) setupGitEnvironment() ([]string, error) {
 		if err != nil {
 			return nil, fmt.Errorf("cannot fix source secret permissions: %v", err)
 		}
-
+		defer os.RemoveAll(sourceSecretDir)
```

Comment:
This is not the appropriate place to remove the source secret directory, since environment variables have been set that point to files in this directory so git clone can succeed. It would need to be removed after the source has been cloned.